code/app/src/main/java/com/example/goldencarrot/views/EntrantHomeView.java

```diff
@@ -277,4 +336,106 @@ private void loadEventData() {
             }
         });
     }
+
+    /**
```

Comment:
This starts to make the logic of this activity a bit too complex, is there any chance we can try to implement these methods in a separate class?